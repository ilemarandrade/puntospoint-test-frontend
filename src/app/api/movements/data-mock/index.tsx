import { faker } from '@faker-js/faker';
import { eachDayOfInterval, format, getMonth, getYear } from 'date-fns';

export const generateWeeklyData = () => {
  const daysOfWeek = [
    'Lunes',
    'Martes',
    'Miércoles',
    'Jueves',
    'Viernes',
    'Sábado',
    'Domingo',
  ];

  const currentDate = new Date();
  const currentDay = currentDate.getDay(); // 0 = Domingo, 1 = Lunes, ..., 6 = Sábado
  const adjustedCurrentDay = (currentDay + 6) % 7; // Ajustar para que Lunes sea 0 y Domingo sea 6

  return daysOfWeek.map((day, index) => {
    // Si el índice del día es mayor que el día actual, establecer valores en 0
    if (index > adjustedCurrentDay) {
      return {
        date: day,
        newCustomers: 0,
        purchased: 0,
        notPurchased: 0,
        totalCustomers: 0,
        totalMoney: 0,
        sales: 0,
        returns: 0,
        cashbackGenerated: 0,
        cashbackAccumulated: 0,
        totalCashback: 0,
        transactions: 0,
      };
    }

    // Generar datos normalmente para los días válidos
    return {
      date: day,
      newCustomers: faker.number.int({ min: 200, max: 400 }),
      purchased: faker.number.int({ min: 1000, max: 3000 }),
      notPurchased: faker.number.int({ min: 200, max: 600 }),
      totalCustomers: faker.number.int({ min: 2000, max: 3000 }),
      totalMoney: faker.number.int({ min: 5000000, max: 12000000 }),
      sales: faker.number.int({ min: 1000000, max: 7000000 }),
      returns: faker.number.int({ min: 200000, max: 600000 }),
      cashbackGenerated: faker.number.int({ min: 100000, max: 300000 }),
      cashbackAccumulated: faker.number.int({ min: 100000, max: 1500000 }),
      totalCashback: faker.number.int({ min: 200000, max: 2000000 }),
      transactions: faker.number.int({ min: 800, max: 2000 }),
    };
  });
};

export const generateMonthlyDataLast6Months = () => {
  const currentDate = new Date();
  const data = [];

  for (let i = 0; i < 6; i++) {
    const month = new Date(
      currentDate.getFullYear(),
      currentDate.getMonth() - i,
      1
    );
    data.push({
      date: month.toLocaleString('es-ES', { month: 'long' }),
      newCustomers: faker.number.int({ min: 200, max: 400 }),
      purchased: faker.number.int({ min: 1000, max: 3000 }),
      notPurchased: faker.number.int({ min: 200, max: 600 }),
      totalCustomers: faker.number.int({ min: 2000, max: 3000 }),
      totalMoney: faker.number.int({ min: 5000000, max: 12000000 }),
      sales: faker.number.int({ min: 1000000, max: 7000000 }),
      returns: faker.number.int({ min: 200000, max: 600000 }),
      cashbackGenerated: faker.number.int({ min: 100000, max: 300000 }),
      cashbackAccumulated: faker.number.int({ min: 100000, max: 1500000 }),
      totalCashback: faker.number.int({ min: 200000, max: 2000000 }),
      transactions: faker.number.int({ min: 800, max: 2000 }),
    });
  }

  return data.reverse(); // Para que los meses estén en orden cronológico
};

export const generateYearlyData = () => {
  const currentYear = new Date().getFullYear();
  const data = [];

  for (let year = 2020; year <= currentYear; year++) {
    data.push({
      date: year.toString(), // Año como string
      newCustomers: faker.number.int({ min: 2000, max: 5000 }),
      purchased: faker.number.int({ min: 10000, max: 30000 }),
      notPurchased: faker.number.int({ min: 2000, max: 6000 }),
      totalCustomers: faker.number.int({ min: 5000, max: 10000 }),
      totalMoney: faker.number.int({ min: 10000000, max: 50000000 }),
      sales: faker.number.int({ min: 1000000, max: 7000000 }),
      returns: faker.number.int({ min: 200000, max: 600000 }),
      cashbackGenerated: faker.number.int({ min: 100000, max: 300000 }),
      cashbackAccumulated: faker.number.int({ min: 100000, max: 1500000 }),
      totalCashback: faker.number.int({ min: 200000, max: 2000000 }),
      transactions: faker.number.int({ min: 800, max: 2000 }),
    });
  }

  return data;
};

export const generateHourlyData = (isToday = false) => {
  const currentDate = new Date();
  const currentHour = currentDate.getHours(); // Obtener la hora actual (0-23)

  const hours = Array.from({ length: 24 }, (_, index) => index); // Genera un array de 0 a 23

  return hours.map((hour) => {
    // Formatear la hora en AM/PM
    const isPM = hour >= 12;
    const formattedHour = hour % 12 || 12; // Convertir a formato de 12 horas
    const amPm = isPM ? 'pm' : 'am';

    // Si es hoy y la hora es mayor que la hora actual, establecer valores en 0
    if (isToday && hour > currentHour) {
      return {
        date: `${formattedHour}:00 ${amPm}`, // Formato HH:MM AM/PM
        newCustomers: 0,
        purchased: 0,
        notPurchased: 0,
        totalCustomers: 0,
        totalMoney: 0,
        sales: 0,
        returns: 0,
        cashbackGenerated: 0,
        cashbackAccumulated: 0,
        totalCashback: 0,
        transactions: 0,
      };
    }

    // Generar datos normalmente para las horas válidas
    return {
      date: `${formattedHour}:00 ${amPm}`, // Formato HH:MM AM/PM
      newCustomers: faker.number.int({ min: 10, max: 100 }),
      purchased: faker.number.int({ min: 5, max: 50 }),
      notPurchased: faker.number.int({ min: 1, max: 20 }),
      totalCustomers: faker.number.int({ min: 100, max: 500 }),
      totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
      sales: faker.number.int({ min: 1000, max: 10000 }),
      returns: faker.number.int({ min: 100, max: 500 }),
      cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
      cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
      totalCashback: faker.number.int({ min: 100, max: 2000 }),
      transactions: faker.number.int({ min: 10, max: 100 }),
    };
  });
};

export const generateAllData = () => {
  return {
    weeklyData: generateWeeklyData(),
    monthlyData: generateMonthlyDataLast6Months(),
    yearlyData: generateYearlyData(),
  };
};

export const generateDailyDataForMonth = (month?: number) => {
  const currentDate = new Date();
  const year = currentDate.getFullYear();
  const currentMonth = currentDate.getMonth(); // Mes actual (0 = enero, 11 = diciembre)
  const currentDay = currentDate.getDate(); // Día actual

  // Si el mes especificado es mayor que el mes actual, devolver un array vacío
  if (month !== undefined && month > currentMonth) {
    return []; // O puedes usar `return null;` si prefieres
  }

  const targetMonth = month !== undefined ? month : currentMonth; // Si se pasa un mes, usarlo; si no, usar el mes actual
  const daysInMonth = new Date(year, targetMonth + 1, 0).getDate(); // Número de días en el mes
  const data = [];

  for (let day = 1; day <= daysInMonth; day++) {
    const date = new Date(year, targetMonth, day);
    if (targetMonth === currentMonth && day > currentDay) {
      break;
    } else {
      // Generar datos normalmente
      data.push({
        date: date.toLocaleDateString('es-ES', {
          month: 'long',
          day: 'numeric',
        }),
        newCustomers: faker.number.int({ min: 50, max: 200 }),
        purchased: faker.number.int({ min: 20, max: 100 }),
        notPurchased: faker.number.int({ min: 5, max: 30 }),
        totalCustomers: faker.number.int({ min: 200, max: 500 }),
        totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
        sales: faker.number.int({ min: 1000, max: 10000 }),
        returns: faker.number.int({ min: 100, max: 500 }),
        cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
        cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
        totalCashback: faker.number.int({ min: 100, max: 2000 }),
        transactions: faker.number.int({ min: 10, max: 100 }),
      });
    }
  }

  return data;
};

export const generateMonthlyData = (year?: number) => {
  const currentDate = new Date();
  const currentYear = currentDate.getFullYear();
  const currentMonth = currentDate.getMonth(); // 0 = enero, 11 = diciembre

  const data = [];
  const targetYear = year || currentYear; // Si no se pasa un año, usar el año actual

  // Si se pasa un año, generar datos para todos los meses de ese año
  for (let month = 0; month < 12; month++) {
    // Si el año es el actual, solo generar hasta el mes actual
    if (targetYear === currentYear && month > currentMonth) {
      break; // Salir del bucle si se pasa el mes actual
    }

    const date = new Date(targetYear, month, 1);
    data.push({
      date: date.toLocaleString('es-ES', { month: 'long', year: 'numeric' }), // Formato: "octubre de 2023"
      newCustomers: faker.number.int({ min: 50, max: 200 }),
      purchased: faker.number.int({ min: 20, max: 100 }),
      notPurchased: faker.number.int({ min: 5, max: 30 }),
      totalCustomers: faker.number.int({ min: 200, max: 500 }),
      totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
      sales: faker.number.int({ min: 1000, max: 10000 }),
      returns: faker.number.int({ min: 100, max: 500 }),
      cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
      cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
      totalCashback: faker.number.int({ min: 100, max: 2000 }),
      transactions: faker.number.int({ min: 10, max: 100 }),
    });
  }

  return data;
};

export const generateMonthlyDataFrom2020To2025 = () => {
  const startYear = 2020;
  const endYear = 2025;
  const data = [];

  for (let year = startYear; year <= endYear; year++) {
    data.push({
      date: year.toString(),
      newCustomers: faker.number.int({ min: 50, max: 200 }),
      purchased: faker.number.int({ min: 20, max: 100 }),
      notPurchased: faker.number.int({ min: 5, max: 30 }),
      totalCustomers: faker.number.int({ min: 200, max: 500 }),
      totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
      sales: faker.number.int({ min: 1000, max: 10000 }),
      returns: faker.number.int({ min: 100, max: 500 }),
      cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
      cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
      totalCashback: faker.number.int({ min: 100, max: 2000 }),
      transactions: faker.number.int({ min: 10, max: 100 }),
    });
  }

  return data;
};

export const generateMonthlyDataByRange = (from: Date, to: Date) => {
  const data = [];

  const startYear = getYear(from);
  const startMonth = getMonth(from); // 0-11
  const endYear = getYear(to);
  const endMonth = getMonth(to); // 0-11

  for (let year = startYear; year <= endYear; year++) {
    const monthStart = year === startYear ? startMonth : 0;
    const monthEnd = year === endYear ? endMonth : 11;

    for (let month = monthStart; month <= monthEnd; month++) {
      const date = new Date(year, month, 1);
      data.push({
        date: date,
        newCustomers: faker.number.int({ min: 50, max: 200 }),
        purchased: faker.number.int({ min: 20, max: 100 }),
        notPurchased: faker.number.int({ min: 5, max: 30 }),
        totalCustomers: faker.number.int({ min: 200, max: 500 }),
        totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
        sales: faker.number.int({ min: 1000, max: 10000 }),
        returns: faker.number.int({ min: 100, max: 500 }),
        cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
        cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
        totalCashback: faker.number.int({ min: 100, max: 2000 }),
        transactions: faker.number.int({ min: 10, max: 100 }),
        invoiced: [
          {
            date: new Date(date.setDate(1)),
            amount: 1000,
          },
          {
            date: new Date(date.setDate(10)),
            amount: 1000,
          },
          {
            date: new Date(date.setDate(20)),
            amount: 1000,
          },
        ],
      });
    }
  }

  return data.reverse();
};
export const generateCustomData = (startDate?: Date, endDate?: Date) => {
  if (!startDate || !endDate) return [];

  if (startDate > endDate) {
    const temp = startDate;
    startDate = endDate;
    endDate = temp;
  }

  const days = eachDayOfInterval({ start: startDate, end: endDate });

  const dailyData = days.map((day) => {
    const formattedDate = format(day, 'dd-MM-yyyy');

    return {
      date: formattedDate,
      newCustomers: faker.number.int({ min: 10, max: 100 }),
      purchased: faker.number.int({ min: 5, max: 50 }),
      notPurchased: faker.number.int({ min: 1, max: 20 }),
      totalCustomers: faker.number.int({ min: 100, max: 500 }),
      totalMoney: faker.number.int({ min: 100000, max: 1000000 }),
      sales: faker.number.int({ min: 1000, max: 10000 }),
      returns: faker.number.int({ min: 100, max: 500 }),
      cashbackGenerated: faker.number.int({ min: 50, max: 300 }),
      cashbackAccumulated: faker.number.int({ min: 50, max: 1500 }),
      totalCashback: faker.number.int({ min: 100, max: 2000 }),
      transactions: faker.number.int({ min: 10, max: 100 }),
    };
  });

  return dailyData;
};
