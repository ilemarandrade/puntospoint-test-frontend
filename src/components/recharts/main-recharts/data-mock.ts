import { ISalesData } from '@/types/recharts';

export const data: ISalesData[] = [
  {
    date: 'Monday',
    newCustomers: 220,
    purchased: 2100,
    notPurchased: 300,
    totalCustomers: 2400,
    totalMoney: 6000000,
    sales: 1000000,
    returns: 500000,
    cashbackGenerated: 150000,
    cashbackAccumulated: 150000,
    totalCashback: 200000,
    transactions: 1800,
  },
  {
    date: 'Tuesday',
    newCustomers: 250,
    purchased: 1100,
    notPurchased: 350,
    totalCustomers: 1450,
    totalMoney: 5000000,
    sales: 2000000,
    returns: 400000,
    cashbackGenerated: 120000,
    cashbackAccumulated: 270000,
    totalCashback: 350000,
    transactions: 950,
  },
  {
    date: 'Wednesday',
    newCustomers: 200,
    purchased: 1500,
    notPurchased: 400,
    totalCustomers: 1900,
    totalMoney: 5500000,
    sales: 3500000,
    returns: 450000,
    cashbackGenerated: 180000,
    cashbackAccumulated: 450000,
    totalCashback: 550000,
    transactions: 1300,
  },
  {
    date: 'Thursday',
    newCustomers: 250,
    purchased: 1900,
    notPurchased: 450,
    totalCustomers: 2350,
    totalMoney: 7000000,
    sales: 6500000,
    returns: 500000,
    cashbackGenerated: 220000,
    cashbackAccumulated: 670000,
    totalCashback: 800000,
    transactions: 1700,
  },
  {
    date: 'Friday',
    newCustomers: 230,
    purchased: 2100,
    notPurchased: 500,
    totalCustomers: 2600,
    totalMoney: 6000000,
    sales: 5000000,
    returns: 550000,
    cashbackGenerated: 250000,
    cashbackAccumulated: 920000,
    totalCashback: 1100000,
    transactions: 1900,
  },
  {
    date: 'Saturday',
    newCustomers: 320,
    purchased: 1050,
    notPurchased: 600,
    totalCustomers: 1650,
    totalMoney: 11000000,
    sales: 10000000,
    returns: 600000,
    cashbackGenerated: 300000,
    cashbackAccumulated: 1220000,
    totalCashback: 1450000,
    transactions: 900,
  },
  {
    date: 'Sunday',
    newCustomers: 330,
    purchased: 1100,
    notPurchased: 550,
    totalCustomers: 1650,
    totalMoney: 10000000,
    sales: 9500000,
    returns: 550000,
    cashbackGenerated: 280000,
    cashbackAccumulated: 1500000,
    totalCashback: 1750000,
    transactions: 950,
  },
];
