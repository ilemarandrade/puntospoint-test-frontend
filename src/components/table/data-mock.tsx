import { Column } from '.';

type Person = {
  firstName: string;
  lastName: string;
  age: number;
  visits: number;
  status: string;
  progress: number;
};

export const defaultData: Person[] = [
  {
    firstName: 'tanner',
    lastName: 'linsley',
    age: 24,
    visits: 100,
    status: 'In Relationship',
    progress: 50,
  },
  {
    firstName: 'tandy',
    lastName: 'miller',
    age: 40,
    visits: 40,
    status: 'Single',
    progress: 80,
  },
  {
    firstName: 'joe',
    lastName: 'dirte',
    age: 45,
    visits: 20,
    status: 'Complicated',
    progress: 10,
  },
];

export const columns: Column<Person>[] = [
  {
    header: 'First Name',
    keyAccessor: 'firstName',
  },
  {
    header: 'Last Name',
    keyAccessor: 'lastName',
    render: (row) => <i>{row.lastName}</i>,
  },
  {
    header: 'Age',
    keyAccessor: 'age',
  },
  {
    header: 'Visits',
    keyAccessor: 'visits',
  },
  {
    header: 'Status',
    keyAccessor: 'status',
  },
  {
    header: 'Profile Progress',
    keyAccessor: 'progress',
  },
];
