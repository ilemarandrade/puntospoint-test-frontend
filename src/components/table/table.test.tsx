import React from 'react';
import { render, screen } from '@testing-library/react';
import Table from '.';
import { columns, defaultData } from './data-mock';

describe('Table Component', () => {
  test('renders the table with correct data', () => {
    const { container } = render(
      <Table
        title="Test Table"
        columns={columns}
        data={defaultData}
        dataTestid="testId"
      />
    );

    expect(screen.getByText('Test Table')).toBeInTheDocument();

    columns.forEach((column) => {
      expect(screen.getByText(column.header)).toBeInTheDocument();
    });

    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    container.querySelectorAll('tbody tr').forEach((row, indexRow) => {
      // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
      row.querySelectorAll('td').forEach((currentColumn, index) => {
        const isRender = !!columns[index].render;
        const plainText = defaultData[indexRow][columns[index].keyAccessor];

        if (isRender) {
          const { container: customRenderFromColumn } = render(
            <>
              {columns[index].render &&
                columns[index].render(defaultData[indexRow] as any)}
            </>
          );

          expect(currentColumn.innerHTML).toContain(
            customRenderFromColumn.innerHTML
          );
        } else {
          expect(currentColumn).toHaveTextContent(plainText.toString());
        }
      });
    });
    expect(container).toMatchSnapshot();
  });
});
