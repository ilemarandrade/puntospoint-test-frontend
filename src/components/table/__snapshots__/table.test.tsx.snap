// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders the table with correct data 1`] = `
<div>
  <div
    class="p-4 bg-[#e6e1e6] rounded-[10px] "
  >
    <h4
      class="text-center text-sm mb-4 font-medium text-[#48454E]"
    >
      Test Table
    </h4>
    <div
      class="MuiTableContainer-root css-70zvr5-MuiTableContainer-root"
      numberofcolumns="6"
    >
      <table
        aria-label="simple table"
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              First Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              Last Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              Age
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              Visits
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              Status
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-bmnu5q-MuiTableCell-root"
              scope="col"
            >
              Profile Progress
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              aria-label="tanner"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              tanner
            </td>
            <td
              aria-label="linsley"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              <i>
                linsley
              </i>
            </td>
            <td
              aria-label="24"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              24
            </td>
            <td
              aria-label="100"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              100
            </td>
            <td
              aria-label="In Relationship"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              In Relationship
            </td>
            <td
              aria-label="50"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              50
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              aria-label="tandy"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              tandy
            </td>
            <td
              aria-label="miller"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              <i>
                miller
              </i>
            </td>
            <td
              aria-label="40"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              40
            </td>
            <td
              aria-label="40"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              40
            </td>
            <td
              aria-label="Single"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              Single
            </td>
            <td
              aria-label="80"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              80
            </td>
          </tr>
          <tr
            class="MuiTableRow-root css-6phqnb-MuiTableRow-root"
          >
            <td
              aria-label="joe"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              joe
            </td>
            <td
              aria-label="dirte"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              <i>
                dirte
              </i>
            </td>
            <td
              aria-label="45"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              45
            </td>
            <td
              aria-label="20"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              20
            </td>
            <td
              aria-label="Complicated"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              Complicated
            </td>
            <td
              aria-label="10"
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium text-nowrap css-14y13q2-MuiTableCell-root"
              data-mui-internal-clone-element="true"
            >
              10
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
